src/test/java/io/pinecone/PineconeTest.java

```diff
@@ -0,0 +1,238 @@
+        when(mockCall.execute()).thenReturn(mockResponse);
+
+        Pinecone client = new Pinecone("testApiKey", mockClient);
+        IndexModel index = client.describeIndex("testIndex");
+
+        assertEquals(expectedIndex, index);
+    }
+
+    @Test
+    public void testListIndexes() throws IOException {
+        String filePath = "src/test/resources/indexListJsonString.json";
+        String indexListJsonString = new String(Files.readAllBytes(Paths.get(filePath)));
+        IndexList expectedIndexList = gson.fromJson(indexListJsonString, IndexList.class);
+
+        Call mockCall = mock(Call.class);
+
+        Response mockResponse = new Response.Builder()
+                .request(new Request.Builder().url("http://localhost").build())
+                .protocol(Protocol.HTTP_1_1)
+                .code(200)
+                .message("OK")
+                .body(ResponseBody.create(indexListJsonString, MediaType.parse("application/json")))
+                .build();
+
+        OkHttpClient mockClient = mock(OkHttpClient.class);
+        when(mockClient.newCall(any(Request.class))).thenReturn(mockCall);
+        when(mockCall.execute()).thenReturn(mockResponse);
+
+        Pinecone client = new Pinecone("testApiKey", mockClient);
+        IndexList indexList = client.listIndexes();
+        assertEquals(indexList, expectedIndexList);
+    }
+
+    @Test
+    public void testConfigureIndex() throws IOException {
+        String filePath = "src/test/resources/podIndexJsonString.json";
+        String podIndexJsonString = new String(Files.readAllBytes(Paths.get(filePath)));
+        IndexModel expectedConfiguredIndex = gson.fromJson(podIndexJsonString, IndexModel.class);
+
+        ConfigureIndexRequestSpecPod pod = new ConfigureIndexRequestSpecPod().podType("s1.x2").replicas(3);
+        ConfigureIndexRequestSpec spec = new ConfigureIndexRequestSpec().pod(pod);
+        ConfigureIndexRequest configureIndexRequest = new ConfigureIndexRequest().spec(spec);
+
+        Call mockCall = mock(Call.class);
+        when(mockCall.execute()).thenReturn(new Response.Builder()
+                .request(new Request.Builder().url("http://localhost").build())
+                .protocol(Protocol.HTTP_1_1)
+                .code(202)
+                .message("The index has been successfully updated.")
+                .body(ResponseBody.create(podIndexJsonString, MediaType.parse("application/json")))
+                .build());
+
+        OkHttpClient mockClient = mock(OkHttpClient.class);
+        when(mockClient.newCall(any(Request.class))).thenReturn(mockCall);
+        Pinecone client = new Pinecone("testApiKey", mockClient);
+        IndexModel configuredIndex = client.configureIndex("testIndex", configureIndexRequest);
+
+        verify(mockClient, times(1)).newCall(any(Request.class));
+        verify(mockCall, times(1)).execute();
+        assertEquals(expectedConfiguredIndex, configuredIndex);
+    }
+}
```

Comment:
Since this file is so long, would it be worth it to break it up into more readable files?

Reply:
It is a bit long, I'm planning on adding more in as well so it probably does make sense to break it into chunks. Let me take a look.

Reply:
Things have been split into:

- `PineconeBuilderTest`
- `PineconeConfigTest`
- `PineconeIndexOperationsTest`